Sources/Query.swift

```diff
@@ -8,159 +8,507 @@
-            return values.joined(separator: ",")
+     
+     */
+    public init(where name: String, _ operation: QueryOperation, for locale: String? = nil) {
+        self.init()
+        self.parameters = [String: String]()
+        self.addFilter(where: name, operation, for: locale)
+    }
+
+    fileprivate func addFilter(where name: String, _ operation: QueryOperation, for locale: String? = nil) {
 
-        case .includes(let values):
-            return values.joined(separator: ",")
+        // Create parameter for this query operation.
+        let parameter = name + operation.string
+        parameters[parameter] = operation.values
+        set(locale: locale)
+    }
 
-        case .excludes(let values):
-            return values.joined(separator: ",")
+    fileprivate func set(locale: String?) {
+        guard let locale = locale else { return }
+        parameters[QueryParameter.locale] = locale
+    }
 
-        case .exists(let value):
-            return value ? "true" : "false"
+    fileprivate init(parameters: [String: String], locale: String?) {
+        self.init()
+        self.parameters = parameters
+
+        if let locale = locale {
+            self.parameters[QueryParameter.locale] = locale
         }
     }
 }
 
-public struct Query<ContentType: ContentModel> {
+/// Protocol which enables concrete query implementations to be 'chained' together so that results
+/// can be filtered by more than one QueryOperation or other query. Protocol extensions give default implementation
+/// so that all concrete types, `Query`, `AssetQuery`, `FilterQuery`, and `QueryOn<EntryType>`, can use the same implementation.
+public protocol ChainableQuery: AbstractQuery {}
+public extension ChainableQuery {
+
+    /**
+     Convenience initializer for a select operation query in which only the fields specified in the fieldNames property will be returned in the JSON response.
+     The `"sys"` dictionary is always requested by the SDK. 
+     Note that if you are using the select operator with an instance `QueryOn<EntryType>`
+     that you must make properties that you are ommitting in the response (by not including them in your selections array) optional properties.
+     Example usage:
+     
+     ```
+     let query = try! QueryOn<Cat>(selectingFieldsNamed: ["fields.bestFriend", "fields.color", "fields.name"])
+     client.fetchEntries(with: query).observable.then { cats in
+        // Do stuff with cats.
+     }
+     ```
+
+     See: <https://www.contentful.com/developers/docs/references/content-delivery-api/#/reference/search-parameters/select-operator>
+     - Parameter fieldNames: An array of field names to include in the JSON response.
+     - Parameter locale: An optional locale argument to return localized results. If unspecified, the locale originally
+     set on the `Client` instance is used.
+     - Throws: Will throw an error if property names are not prefixed with `"fields."`, if selections go more than 2 levels deep
```

Comment:
The names do not need to be prefixed with fields. ContentTypes and Assets for instance do have their name on the top level, without the need of using `fields.`.